src/main/java/hudson/plugins/gradle/enriched/ScanDetailService.java

```diff
@@ -0,0 +1,139 @@
+            return Optional.ofNullable(doGetScanDetail(buildScanUrl));
+        }
+
+        return Optional.empty();
+    }
+
+    private ScanDetail doGetScanDetail(String buildScanUrl) {
+        if (buildScanUrl.startsWith(GRADLE_ENTERPRISE_PUBLIC_SERVER)) {
+            // API is not accessible on public server
+            return null;
+        }
+
+        String baseApiUri = getBaseApiUri(buildScanUrl);
+        if (null == baseApiUri || baseApiUri.isEmpty()) {
+            LOGGER.error("Gradle Enterprise API URL can't be resolved");
+            return null;
+        }
+
+        try (CloseableHttpClient httpclient = httpClientFactory.buildHttpClient(httpClientTimeoutInSeconds, httpClientMaxRetries, httpClientDelayBetweenRetriesInSeconds)) {
+            HttpGet httpGetApiBuilds = buildGetRequest(baseApiUri);
+
+            ScanDetail scanDetail = new ScanDetail(buildScanUrl);
+            try (CloseableHttpResponse responseApiBuilds = httpclient.execute(httpGetApiBuilds)) {
+                if (responseApiBuilds.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                    LOGGER.warn("Unable to fetch build scan data [{}]", responseApiBuilds.getStatusLine().getStatusCode());
+                    return null;
+                }
+
+                HttpEntity httpEntityApiBuilds = responseApiBuilds.getEntity();
+                if (httpEntityApiBuilds != null) {
+                    String apiBuildsResponse = EntityUtils.toString(httpEntityApiBuilds);
+                    ObjectReader objectReader = MAPPER.readerForUpdating(scanDetail);
+                    scanDetail = objectReader.readValue(apiBuildsResponse);
+                    String suffix = null != scanDetail.getBuildToolType() ? scanDetail.getBuildToolType().getAttributesUrlSuffix() : "unsupported";
+                    HttpGet httpGetBuildAttributes = buildGetRequest(baseApiUri + suffix);
+
+                    try (CloseableHttpResponse responseApiBuildAttributes = httpclient.execute(httpGetBuildAttributes)) {
+                        if (responseApiBuildAttributes.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                            LOGGER.warn("Unable to fetch build scan data [{}]", responseApiBuildAttributes.getStatusLine().getStatusCode());
+                            return null;
+                        }
+
+                        HttpEntity httpEntityApiBuildAttributes = responseApiBuildAttributes.getEntity();
+                        if (httpEntityApiBuildAttributes != null) {
+                            String apiBuildAttributesResponse = EntityUtils.toString(httpEntityApiBuildAttributes);
+                            scanDetail = objectReader.readValue(apiBuildAttributesResponse);
+                            return scanDetail;
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.info("Error fetching data [{}]", e.getMessage());
```

Comment:
It's better to log with the stack trace, so we can have more info in case we need to investigate it.

```suggestion
            LOGGER.warn("Error fetching build scan details", e);
```